src/main/java/com/conveyal/gtfs/GTFSFeed.java

```diff
@@ -397,6 +395,23 @@ public Shape getShape (String shape_id) {
         return shape.shape_dist_traveled.length > 0 ? shape : null;
     }
 
+    /**
+     * MapDB-based implementation to find patterns.
+     *
+     * FIXME: Remove and make pattern finding happen during validation?p
```

Comment:
I believe this comment sums up the reason the original method was removed. I wanted to share the pattern finder between the two implementations (MapDB and RDBMS), apply the same validation process to both kinds of storage, and produce Patterns in the same way in both cases, during validation. This prevents us from iterating over every stopTime twice, since we're already iterating over all of them in validation. This comment should probably be expanded to explain the intended path of development.

Reply:
OK, I will expand the comment to include this.  It's worth noting, however, that the stop times for MapDB have already been placed in a map where it is easy to retrieve them.